frontend/src/metabase/actions/containers/ActionCreator/ActionCreatorView.tsx

```diff
@@ -107,29 +106,23 @@ export default function ActionCreatorView({
               />,
             ].filter(isNotNull)}
           />
-          <EditorContainer>
-            <QueryActionEditor
-              query={question.query() as NativeQuery}
-              isEditable={isEditable}
-              onChangeQuestionQuery={onChangeQuestionQuery}
-            />
-          </EditorContainer>
+          <EditorContainer>{children}</EditorContainer>
           <ModalActions>
             <Button onClick={onCloseModal} borderless>
               {t`Cancel`}
             </Button>
             {isEditable && (
-              <Button primary disabled={canSave} onClick={onClickSave}>
+              <Button primary disabled={!canSave} onClick={onClickSave}>
                 {isNew ? t`Save` : t`Update`}
               </Button>
             )}
           </ModalActions>
         </ModalLeft>
         {activeSideView === "actionForm" ? (
           <FormCreator
-            params={question.parameters() ?? []}
+            params={action.parameters ?? []}
             formSettings={formSettings}
-            isEditable={isEditable}
+            isEditable={isEditable && canChangeFieldSettings}
```

Comment:
should we rename these variables to be a bit more conveying? I checked `isEditable` checks if the model's DB has actions enabled. And `canChangeFieldSettings` comes from action contexts. Maybe they could name something like `isActionEnabled` + `isActionEditable`. Not totally sure if that will help but `isEditable` + `canChangeFieldSettings` doesn't sound as related.